aioxmpp/avatar/service.py

```diff
@@ -391,17 +440,146 @@ class AvatarService(service.Service):
 
     def __init__(self, client, **kwargs):
         super().__init__(client, **kwargs)
-        self._has_pep = None
-        self._metadata_cache = {}
+        self._has_pep_avatar = set()
+        self._metadata_cache = LRUDict()
+        self._metadata_cache.maxsize = 200
         self._pubsub = self.dependencies[pubsub.PubSubClient]
-        self._notify_lock = asyncio.Lock()
+        self._pep = self.dependencies[pep.PEPClient]
+        self._presence_server = self.dependencies[presence.PresenceServer]
         self._disco = self.dependencies[disco.DiscoClient]
+        self._vcard = self.dependencies[vcard.VCardService]
         # we use this lock to prevent race conditions between different
         # calls of the methods by one client.
         # XXX: Other, independent clients may still cause inconsistent
         # data by race conditions, this should be fixed by at least
         # checking for consistent data after an update.
         self._publish_lock = asyncio.Lock()
+        self._synchronize_vcard = False
+        self._vcard_ressource_interference = set()
+        self._vcard_id = None
+        self._vcard_rehash_task = None
+
+    @property
+    def metadata_cache_size(self):
+        """
+        Maximum number of cache entries in the avatar metadata cache.
+
+        This is mostly a measure to prevent malicious peers from
+        exhausting memory by spamming vCard based avatar metadata for
+        different ressources.
+
+        .. versionadded:: 0.10
+
+        """
+        return self._metadata_cache.maxsize
+
+    @metadata_cache_size.setter
+    def metadata_cache_size(self, value):
+        self._metadata_cache.maxsize = value
+
+    @property
+    def synchronize_vcard(self):
+        """
+        Set this property to true to enable publishing the a vCard avatar.
+
+        This property defaults to false. For the setting true to have
+        effect, you have to publish your avatar with :meth:`publish_avatar_set`
+        or :meth:`disable_avatar` *after* this switch has been set to true.
+        """
+        return self._synchronize_vcard
+
+    @synchronize_vcard.setter
+    def synchronize_vcard(self, value):
+        self._synchronize_vcard = bool(value)
+
+    @service.depfilter(aioxmpp.stream.StanzaStream,
+                       "service_outbound_presence_filter")
+    def _attach_vcard_notify_to_presence(self, stanza):
+        if not self._vcard_ressource_interference:
+            stanza.xep0153_x = avatar_xso.VCardTempUpdate(self._vcard_id)
+        return stanza
+
+    def _handle_notify(self, full_jid, stanza):
+        if stanza.xep0153_x is not None:
```

Comment:
```python
if stanza.x0153_x is None:
    return
```

Saves a whole level of indentation on the remaining method. Same for the next ``if``.